modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/related/models/test/ObjectRelatedModelsProviderTest.java

```diff
@@ -957,6 +946,103 @@ private void _testSystemObjectEntry1toMObjectRelatedModels()
+
+		CompanyThreadLocal.setCompanyId(companyId);
+		PermissionThreadLocal.setPermissionChecker(
+			PermissionCheckerFactoryUtil.create(user));
+		PrincipalThreadLocal.setName(user.getUserId());
+
+		ObjectDefinition objectDefinition = _addObjectDefinition();
+
+		ObjectDefinition systemObjectDefinition =
+			_objectDefinitionLocalService.fetchObjectDefinitionByClassName(
+				companyId, AccountEntry.class.getName());
+
+		ObjectRelationship objectRelationship = _addObjectRelationship(
+			objectDefinition.getObjectDefinitionId(),
+			systemObjectDefinition.getObjectDefinitionId(),
+			ObjectRelationshipConstants.TYPE_ONE_TO_MANY);
+
+		AccountEntry accountEntry1 = _addAccountEntry(user.getUserId());
+		AccountEntry accountEntry2 = _addAccountEntry(user.getUserId());
+
+		ObjectEntry objectEntry = _objectEntryLocalService.addObjectEntry(
+			user.getUserId(), 0, objectDefinition.getObjectDefinitionId(),
+			Collections.emptyMap(), ServiceContextTestUtil.getServiceContext());
+
+		String relationshipColumnName = StringBundler.concat(
+			"r_", objectRelationship.getName(), "_",
+			objectDefinition.getPKObjectFieldName());
+
+		_objectEntryLocalService.
+			addOrUpdateExtensionDynamicObjectDefinitionTableValues(
+				user.getUserId(), systemObjectDefinition,
+				accountEntry1.getAccountEntryId(),
+				HashMapBuilder.<String, Serializable>put(
+					relationshipColumnName, objectEntry.getObjectEntryId()
+				).build(),
+				ServiceContextTestUtil.getServiceContext());
+
+		ObjectRelatedModelsProvider<ObjectEntry> objectRelatedModelsProvider =
+			_objectRelatedModelsProviderRegistry.getObjectRelatedModelsProvider(
+				systemObjectDefinition.getClassName(),
+				systemObjectDefinition.getCompanyId(),
+				ObjectRelationshipConstants.TYPE_ONE_TO_MANY);
+
+		List<ObjectEntry> unrelatedObjectEntries =
+			objectRelatedModelsProvider.getUnrelatedModels(
+				companyId, 0, systemObjectDefinition,
+				objectEntry.getObjectEntryId(),
+				objectRelationship.getObjectRelationshipId());
+
+		Assert.assertEquals(
+			unrelatedObjectEntries.toString(), 1,
+			unrelatedObjectEntries.size());
+
+		_objectEntryLocalService.
+			addOrUpdateExtensionDynamicObjectDefinitionTableValues(
+				user.getUserId(), systemObjectDefinition,
+				accountEntry2.getAccountEntryId(),
+				HashMapBuilder.<String, Serializable>put(
+					relationshipColumnName, objectEntry.getObjectEntryId()
+				).build(),
+				ServiceContextTestUtil.getServiceContext());
+
+		unrelatedObjectEntries = objectRelatedModelsProvider.getUnrelatedModels(
+			companyId, 0, systemObjectDefinition,
+			objectEntry.getObjectEntryId(),
+			objectRelationship.getObjectRelationshipId());
+
+		Assert.assertEquals(
+			unrelatedObjectEntries.toString(), 0,
+			unrelatedObjectEntries.size());
+
+		_objectRelationshipLocalService.deleteObjectRelationship(
+			objectRelationship.getObjectRelationshipId());
+		_objectDefinitionLocalService.deleteObjectDefinition(
+			objectDefinition.getObjectDefinitionId());
```

Comment:
@murilostodolni could you add an empty line as we need to first delete the relationships before deleting the objects??